crates/lib3h/src/transport/websocket/actor.rs

```diff
@@ -8,35 +8,72 @@ use crate::transport::{
 };
 use detach::Detach;
 use holochain_tracing::Span;
-use lib3h_discovery::{error::DiscoveryResult, Discovery};
+use lib3h_discovery::{
+    error::{DiscoveryError, DiscoveryResult},
+    Discovery,
+};
 use lib3h_ghost_actor::prelude::*;
 use lib3h_protocol::{data_types::Opaque, Address};
 use url::Url;
 
+// Use mDNS for bootstrapping
+use lib3h_mdns::{MulticastDns, MulticastDnsBuilder};
+
 pub type Message =
     GhostMessage<RequestToChild, RequestToParent, RequestToChildResponse, TransportError>;
 
 pub struct GhostTransportWebsocket {
     #[allow(dead_code)]
     machine_id: Address,
+    network_id_address: Address,
     endpoint_parent: Option<GhostTransportWebsocketEndpoint>,
     endpoint_self: Detach<GhostTransportWebsocketEndpointContext>,
     streams: StreamManager<std::net::TcpStream>,
     bound_url: Option<Url>,
     pending: Vec<Message>,
+    mdns: Option<MulticastDns>,
 }
 
+// Here we just need to use mDNS, but use it only once, with advertise probably, and that's all.
 impl Discovery for GhostTransportWebsocket {
     fn advertise(&mut self) -> DiscoveryResult<()> {
-        Ok(())
+        // Lazily instantiate mDNS
+        if self.mdns.is_none() {
```

Comment:
Do we want need to have an instance of the MDNS per memory transport, or can they all access one instance that lives in a mutex in a lazy static?

Reply:
We could do some kind of "discovery manager" concept. Some of the transports can share discovery mechanisms - i.e. native and websocket can both use mdns, while memory and something like bluetooth would probably need specialized mechanisms...

But we can also go with one discovery per transport, because I also think it's totally valid for say native and wss to each have their own mdns queries / responses.

Reply:
I agree with @neonphog , I think it's valid and it can avoid confusion as one shared mDNS instance in a lazy_static would return every Url for every instance of lib3h, instead of doing it only for its own network ID.